core/sdk/src/main/java/com/mongodb/stitch/core/internal/net/AppMetadata.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright 2018-present MongoDB, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.mongodb.stitch.core.internal.net;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mongodb.stitch.core.internal.common.StitchObjectMapper;
+
+/**
+ * Represents the application metadata for a particular Stitch
+ * application.
+ */
+@SuppressWarnings("unused") // Jackson uses reflection
+public class AppMetadata {
```

Comment:
[nit] in our Java code we generally prefix our classes that represent API models with `Api`, so this could be renamed `ApiAppMetadata`